### PR TITLE
Create helper function that returns the actual size of object in memory

### PR DESCRIPTION
**Pull request by Hassan754**

### CODE REVIEW

**Review by faizan2590 — APPROVED** [review]
LGTM

File: d3a_interface/utils.py [hunk 1]
```diff
@@ -253,3 +256,21 @@ def iterate_over_all_modules(modules_path):
...
+
+def deep_size_of(input_obj):
+    """
+    Gets the real size of a python object taking into consideration the nested objects
+    """
+    memory_size = 0
+    ids = set()
+    objects = [input_obj]
+    while objects:
+        new = []
```
Comment (hannesdiedrich) [inline d3a_interface/utils.py:269]: Minor, but how about renaming `new` to something like `child_objects` ?
AFAIU, these are the child objects, e.g. an `Offer` object inside a dict tree, right?

**Review by hannesdiedrich — APPROVED** [review]
One minor comment. LGTM

File: d3a_interface/utils.py (hunk 1 above)
Comment (Hassan754) [reply d3a_interface/utils.py:269]: @hannesdiedrich actually they are the referents objects not the direct children ,
we can name the new to referents_objs if you agree

Comment (hannesdiedrich) [reply d3a_interface/utils.py:269]: Thanks! yes, would be good.

**Pull request closed without merging**